lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -320,6 +320,11 @@ static bool hasUnhandledError(ArrayRef<ASTNode> Nodes) {
   });
 }
 
+static bool
+hasNodeThat(ArrayRef<ASTNode> Nodes, llvm::function_ref<bool(ASTNode)> Pred) {
```

Comment:
If that's the case, I still prefer it extracted to avoid typing `begin()` and `end()`.